chronos/asyncfutures2.nim

```diff
@@ -317,6 +317,8 @@ proc futureContinue*(fut: FutureBase) {.raises: [], gcsafe.} =
       # `await` typically) or completes / fails / is cancelled
       next = fut.internalClosure(fut)
       if fut.internalClosure.finished(): # Reached the end of the transformed proc
+        #TODO hacky
```

Comment:
this needs a comment explaining why it's important that the future remains unfinished until the iterator is done (ie to run `finally` code fully, or really, any code after the last `yield`) - it's not actually hacky, it follows the way the compiler generates code for result - ie `return xxx` is typically rewritten as `result = xxx; return` - it's nuts but here we are.

Reply:
Done